src/Microsoft.Sbom.Parsers.Spdx30SbomParser/Parser/SPDX30Parser.cs

```diff
@@ -157,7 +163,299 @@ public SPDX30Parser(
+            var packageSpdxId = packageElement.SpdxId;
+
+            var packageHasSha256Hash = packageElement.VerifiedUsing.
+                Any(packageVerificationCode => packageVerificationCode.Algorithm ==
+                HashAlgorithm.sha256);
+
+            if (!packageHasSha256Hash)
+            {
+                throw new ParserException($"SBOM document is not NTIA compliant because package with SPDX ID {packageSpdxId} does not have a SHA256 hash.");
+            }
+        }
+    }
+
+    /// <summary>
+    /// Sets metadata based on parsed SBOM elements.
+    /// </summary>
+    /// <param name="result"></param>
+    public SpdxMetadata SetMetadata(ElementsResult result)
+    {
+        var metadata = new SpdxMetadata();
+        var spdxDocumentElement = (SpdxDocument?)result.Elements.FirstOrDefault(element => element.Type == "SpdxDocument");
+
+        if (spdxDocumentElement == null)
+        {
+            return metadata;
+        }
+
+        if (spdxDocumentElement.NamespaceMap != null && spdxDocumentElement.NamespaceMap.TryGetValue("sbom", out var namespaceUri))
+        {
+            metadata.DocumentNamespace = new Uri(namespaceUri);
+        }
+
+        metadata.Name = spdxDocumentElement.Name;
+        metadata.SpdxId = spdxDocumentElement.SpdxId;
+        metadata.DocumentDescribes = spdxDocumentElement.RootElement;
+
+        var dataLicenseSpdxId = spdxDocumentElement.DataLicense;
+        var spdxDataLicenseElement = result.Elements.
+            FirstOrDefault(element => element.SpdxId == dataLicenseSpdxId) as AnyLicenseInfo;
+        metadata.DataLicense = spdxDataLicenseElement?.Name;
+
+        var spdxCreationInfoElement = (CreationInfo?)result.Elements.
+            Where(element => element.Type == nameof(CreationInfo)).
+            FirstOrDefault(element => ((CreationInfo)element).Id == spdxDocumentElement.CreationInfoDetails);
+
+        var creators = spdxCreationInfoElement?.CreatedBy
+            .Select(createdBy => result.Elements.
+            FirstOrDefault(element => element.SpdxId == createdBy) as Organization)
+            .Where(spdxOrganizationElement => spdxOrganizationElement != null)
+            .Select(spdxOrganizationElement => spdxOrganizationElement?.Name)
+            .ToList() ?? [];
+
+        creators.AddRange(spdxCreationInfoElement?.CreatedUsing
+            .Select(createdBy => result.Elements.
+            FirstOrDefault(element => element.SpdxId == createdBy) as Tool)
+            .Where(spdxToolElement => spdxToolElement != null)
+            .Select(spdxToolElement => spdxToolElement?.Name) ?? []);
+
+        var createdDate = DateTime.MinValue;
+        DateTime.TryParse(spdxCreationInfoElement?.Created, out createdDate);
+
+        metadata.CreationInfo = new MetadataCreationInfo
+        {
+            Created = createdDate,
+            Creators = creators,
+        };
+
+        metadata.SpdxVersion = spdxCreationInfoElement?.SpecVersion;
+
+        return metadata;
```

Comment:
This can set multiple properties of `SpdxMetadata` to `null`, when they're not declared as nullable in `SpdxMetadata`. .NET 8.0 assumes non-nullable types unless we explicitly tell it otherwise

Reply:
If result has null values then SpdxMetadata should also have those values set to null, so this is intentional

Reply:
If fields of `SpdxMetadata` can legitimately be null, then we should call them out as nullable in the type. That will cause the compiler to check for places where we might be referencing them without an appropriate null check